bdpy/recon/torch/modules/generator.py

```diff
@@ -0,0 +1,295 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+
+from typing import Callable, Iterator
+
+import torch
+import torch.nn as nn
+from torch.nn.parameter import Parameter
+from bdpy.dl.torch.domain import Domain, image_domain
+
+
+def _get_reset_module_fn(module: nn.Module) -> Callable[[], None] | None:
```

Comment:
This implementation of `_get_reset_module_fn(module)` is dangerous because `nn.Module` does not necessarily implement a parameter resetting function with a predefined name like `reset_parameters`. Actually, typical layers like `nn.Linear` and `nn.Conv2d` have the function `reset_parameters` but `nn.MultiheadAttention` has `_reset_parameters`. We have adopted the conventional rule seen in the official implementation in PyTorch, but
- 3rd party modules do not need to respect this rule
- the official implementation could be changed in the future in the first place

If you have another implementation idea, we welcome it.